Modules/rat_board.py

```diff
@@ -199,10 +199,50 @@ def find_by_uuid(self, guid: UUID) -> Optional[Rescue]:
 
         """
         for rescue in self.rescues.values():
-            if rescue.case_id == guid:
+            if rescue.uuid == guid:
                 return rescue
         return None
 
+    def search(self, case: Union[str, int, UUID]) -> Optional[Rescue]:
+        """
+        Helper for searching for a case using implemented search functions.
+
+        Args:
+            case (Union[str, int, UUID]): the client's name, a case ID, or a rescue UUID
+
+        Returns:
+            Rescue: found rescue
+            None:   No rescue found
+
+        Raises:
+            ValueError: argument was not an expected type
+        """
+        # ensure a proper type was passed
+        if not isinstance(case, str) and not isinstance(case, int) and not isinstance(case, UUID):
+            raise ValueError
+
+        if isinstance(case, str):
+            # strings could be any of the valid types, check for the best one.
+            if case.isdigit():
+                    return self.find_by_index(int(case))
+            else:
+                # UUIDs always start with @ and are 37 characters, so try to parse it
+                if case[0] == "@" and len(case) == 37:
+                    try:
+                        guid_str = case[1:]
+                        guid = UUID(guid_str)
+                    except (AttributeError, ValueError):
+                        # not a valid UUID
+                        return None
```

Comment:
methinks you are writing too defensively. the only time this function should return None is if a rescue was not found. an invalidly formatted uuid should be raised.

Reply:
This function is considered "untrusted" because it will be taking user input, so it will need to deal with all sorts of issues with input. If you want it to raise an exception and make the caller deal with it, I'm on board with that.

Reply:
indeed this function is accepting untrusted user input, but its not responsible to what happens when the data it is given is invalid. I feel the correct course of action is to leave the raised exception to be handled by the caller - the command context.

 Since this method exists outside the command context there is no easy way beyond exceptions to express the fact something went wrong back to the user. Therefore silently handling the error here (defensive programming) makes it far more difficult to implement proper error handling